pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -102,15 +102,24 @@ func (cb *ClusterBuilder) applyDestinationRule(c *cluster.Cluster, clusterMode C
 		// clusters with discovery type STATIC, STRICT_DNS rely on cluster.LoadAssignment field.
 		// ServiceEntry's need to filter hosts based on subset.labels in order to perform weighted routing
 		var lbEndpoints []*endpoint.LocalityLbEndpoints
-		if c.GetType() != cluster.Cluster_EDS {
+
+		isPassthrough := subset.GetTrafficPolicy().GetLoadBalancer().GetConsistentHash() == nil &&
```

Comment:
Why need check this

Reply:
It is not truly needed but just added for safety that user has not specified consistent hash because when we build the cluster we override it with consistent hash https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster.go#L824 because GetSimple has default.  

If you think it is redundant, I can remove it. Let me know

Reply:
Thinking  about this more - I think it is very safe to  remove that condition.

Reply:
https://github.com/istio/istio/pull/29547 - Fix